### DISCUSSION TITLE
A question about double head implementation

### DISCUSSION THREAD

**Discussion by HB-X** [open-mmlab/mmdetection#9035] [Q&A] [answered]
I noticed that the roi used in the classification branch and the regression branch are different. The roi used in the regression branch is larger than the one used in the classification branch because of a scaling factor added to the configuration file. What is the reason for this? Does this have a significant impact on the final detection accuracy? Thank you.
https://github.com/open-mmlab/mmdetection/blob/9d3e162459590eee4cfc891218dfbb5878378842/configs/double_heads/dh_faster_rcnn_r50_fpn_1x_coco.py#L5
https://github.com/open-mmlab/mmdetection/blob/9d3e162459590eee4cfc891218dfbb5878378842/mmdet/models/roi_heads/double_roi_head.py#L21

**Comment by zachcoleman** [comment, marked as answer]
This idea appears to come from the official implementation [here](https://github.com/wuyuebupt/doubleheadsrcnn/search?q=ROI_EXPAND).

My best guess is: this gives the convolutional head more information rather than padding zeros or some other argument about additional spatial information.

**Reply by HB-X** [reply]
@zachcoleman Thank you for your help. In the officical implementation, I found the explanation given by the author.
https://github.com/wuyuebupt/doubleheadsrcnn/issues/2#issuecomment-658496147
Intuitively, I think this strategy has a significant impact on the detection accuracy. I will do a comparison experiment as soon as possible.

**Comment by HB-X** [comment]
I retrained double head by setting reg_roi_scale_factor=None, but the accuracy is much lower than the original one (39.3 AP vs 40.0 AP). The result demonstrates that the strategy of expanding roi does have a very significant impact on the detection accuracy. For fair comparison, I think it is better to specify this trick in configuration file.

**Comment by ZwwWayne** [comment]
This setting is part of the paper's novelty. But you are right, it would be better to add some comments to explain it.
